libs/types/src/investments.rs

```diff
@@ -52,3 +57,197 @@ where
+	/// of an investment
+	/// * If investment: In payment currency
+	/// * If payout: In denomination currency
+	pub payout_investment_invest: Balance,
+
+	/// This is the remaining investment in the payment currency
+	/// of an investment
+	/// * If investment: In payment currency
+	/// * If payout: In denomination currency
+	pub remaining_investment_invest: Balance,
+}
+
+impl<Balance: Zero> Default for InvestCollection<Balance> {
+	fn default() -> Self {
+		InvestCollection {
+			payout_investment_invest: Zero::zero(),
+			remaining_investment_invest: Zero::zero(),
+		}
+	}
+}
+
+impl<Balance: Zero + Copy> InvestCollection<Balance> {
+	/// Create a `InvestCollection` directly from an active invest order of
+	/// a user.
+	/// The field `remaining_investment_invest` is set to the
+	/// amount of the active invest order of the user and will
+	/// be subtracted from upon given fulfillment's
+	pub fn from_order(order: &Order<Balance, OrderId>) -> Self {
+		InvestCollection {
+			payout_investment_invest: Zero::zero(),
+			remaining_investment_invest: order.amount(),
+		}
+	}
+}
+
+/// The outstanding collections for an account
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct RedeemCollection<Balance> {
+	/// This is the payout in the payment currency
+	/// of an investment
+	/// * If redemption: In denomination currency
+	/// * If payout: In payment currency
+	pub payout_investment_redeem: Balance,
+
+	/// This is the remaining redemption in the denomination currency
+	/// of an investment
+	/// * If redemption: In denomination currency
+	/// * If payout: In payment currency
+	pub remaining_investment_redeem: Balance,
+}
+
+impl<Balance: Zero> Default for RedeemCollection<Balance> {
+	fn default() -> Self {
+		RedeemCollection {
+			payout_investment_redeem: Zero::zero(),
+			remaining_investment_redeem: Zero::zero(),
+		}
+	}
+}
+
+impl<Balance: Zero + Copy> RedeemCollection<Balance> {
+	/// Create a `RedeemCollection` directly from an active redeem order of
+	/// a user.
+	/// The field `remaining_investment_redeem` is set to the
+	/// amount of the active redeem order of the user and will
+	/// be subtracted from upon given fulfillment's
+	pub fn from_order(order: &Order<Balance, OrderId>) -> Self {
+		RedeemCollection {
+			payout_investment_redeem: Zero::zero(),
+			remaining_investment_redeem: order.amount(),
+		}
+	}
+}
+
+/// The collected investment/redemption amount for an account
+#[derive(Encode, Default, Decode, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct CollectedAmount<Balance: Default + MaxEncodedLen> {
+	/// The amount which was was collected
+	/// * If investment: Tranche tokens
+	/// * If redemption: Payment currency
+	pub amount_collected: Balance,
+
+	/// The amount which invested and converted during processing based on the
+	/// fulfillment price(s)
+	/// * If investment: Payment currency
+	/// * If redemption: Tranche tokens
+	pub amount_payment: Balance,
+}
```

Comment:
Is this used somewhere?

Reply:
Yes for the return result type of `do_collect_invest` and `do_collect_redeem`, [see my comment](https://github.com/centrifuge/centrifuge-chain/pull/1473#discussion_r1299797271) (I made in between your review as it seems 😅).